app/controllers/api/grade_entry_forms_controller.rb

```diff
@@ -144,7 +144,7 @@ def update_grades
       grade_entry_form = record
 
       grade_entry_student = grade_entry_form.grade_entry_students
-                                            .joins(:user)
+                                            .joins(role: :user)
```

Comment:
I don't have a strong opinion, but to be honest I think it would be more robust to add a `through` association to `GradeEntryStudent` itself. Otherwise we may keep running into this issue for other joins in the future.

Reply:
You're right, it would be - I added the association and updated the places where this pattern was being used for `grade_entry_students`. The same pattern is being used in `memberships`, and could be changed the same way. I will make a separate PR to update that as well.